### PR TITLE
Add fetch on overview

### PR DESCRIPTION
**Pull request by J-Tanaka**
Add fecth on overview menu. It is just do "git fetch."

### CODE REVIEW

**Comment by Andr3as** [comment]
Good idea. It would be nice if you would add an option to fetch a specific remote. "When no remote is specified, by default the origin remote will be used [...]." [[git-scm.com](https://git-scm.com/docs/git-fetch)]

**Comment by J-Tanaka** [comment]
I added an option to fetch a specific remote.

**Comment by J-Tanaka** [comment]
Fix to be able to fetch a default remote.

File: controller.php [hunk 1]
```diff
@@ -275,6 +275,14 @@
                 echo '{"status":"error","message":"Missing parameter!"}';
             }
             break;
+        
+        case 'fetch':
+            if (isset($_GET['path']) && isset($_GET['remote'])) {
+                echo $git->fetch(getWorkspacePath($_GET['path'], $_GET['remote']));
```
Comment (Andr3as) [inline controller.php]: Please correct the parentheses:
`echo $git->fetch(getWorkspacePath($_GET['path']), $_GET['remote']);`

File: class.git.php [hunk 2]
```diff
@@ -343,6 +343,34 @@ public function pull($path, $remote, $branch) {
...
+            if (!is_dir($path)) return $this->returnMessage("error", "Wrong path!");
+            if (!$this->checkExecutableFile()) {
+                return $this->returnMessage("error","Failed to change permissions of shell program");
+            }
+            if (!$this->checkShellProgramExists()) {
+                return $this->returnMessage("error", "Please install shell program!");
+            }
+            
+            $program = $this->getShellProgram();
+            if($remote == 0){
```
Comment (Andr3as) [inline class.git.php]: Does not work for me. After I changed it to `if($remote == "0"){` it worked properly. Could you please test this.

File: dialog.php [hunk 3]
```diff
@@ -19,6 +19,9 @@
             case 'diff':
                 include('templates/diff.html');
                 break;
+            case 'fetch':
+                include('templates/fetch.html');
```
Comment (Andr3as) [inline dialog.php]: This works, but why create a new dialog? Better implement the logic into the [remotes dialog](https://github.com/Andr3as/Codiad-CodeGit/blob/master/templates/remote.html).

File: controller.php (hunk 1 above)
Comment (J-Tanaka) [inline controller.php]: Sorry, I will soon fix the bug and push the commit.

File: dialog.php (hunk 3 above)
Comment (J-Tanaka) [inline dialog.php]: I will Implement into the remotes dialog (and also for overview dialog for <default>.)

File: class.git.php (hunk 2 above)
Comment (J-Tanaka) [inline class.git.php]: It also worked on my environment after changed. I will push the commit soon.

**Comment by J-Tanaka** [comment]
Fixed bugs and Implement fetch into overview and remote.

**Comment by Andr3as** [comment]
Great work. Please resolve the merge conflict.

**Comment by J-Tanaka** [comment]
I have resolved the conflict.

**Pull request merged**